woocommerce/woocommerce-android
# Orders list with last update

⚠️ This PR depends on this [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/3056)

Closes: #11796
Closes: #11799

### Why
This PR shows the latest order sync time in the orders list screen and refresh the first page only when the data is outdated.

### Description
This pull request introduces a mechanism to prevent automatic order list syncing if less than 30 minutes have passed since the last successful sync. This optimization aims to:

- Reduce unnecessary API calls: By preventing frequent syncs, we can minimize API load and potentially improve overall performance.
- Optimize battery life: Fewer background operations can contribute to extended battery life on user's devices.

Implementation Details:

- The last order list sync timestamp will be stored in device preferences.
- Before initiating a new automatic sync, the code will check the time difference between the current time and the stored timestamp.
- If the difference is less than 30 minutes, the automatic sync will be skipped.

Benefits:

- Improved application performance
- Reduced API calls
- Potentially extended battery life

Additional Notes: 

- Users can still manually trigger a sync even if 30 minutes haven't passed by using the pull-to-refresh gesture.

### Testing

TC1 - Opening the order list for the first time
1. Open the app
2. Navigate to the orders section
3. Check that the orders list is displayed successfully
4. Check that the last update section is update with the current time

TC2 - Opening the order list with cached data
1. Run TC1
2. Close the app and wait some minutes
3. Open the app
4. Navigate to the orders section
5. Check that the last update section is displayed with the timestamp from TC1
6. Check that the data is displayed from the DB and no additional request are made to the API to display the first page
7. Pull to refresh
8. Check that the app refresh data from the API
9. Check that the last update section is updated 

TC3 - Filters also works with cached data
1. Open the app
2. Navigate to the orders section
3. Check that the orders list is displayed successfully
4. Tap on Filters
5. Set a filter and press Show Orders
6. Check that the expected orders are displayed
7. Go back to Filters and clear all the filters
8. Check that the orders list data is displayed from the DB without requests to the API 
9. Go back to Filters
10. Set again the filter from step 5
11. Check that the orders list data is displayed from the DB without requests to the API 

TC4
1. Check that searching orders keeps working as expected 

### Images/gif

https://github.com/user-attachments/assets/ce9c4c4a-a838-4a39-8b8e-9da157b64e89
